src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateCopy = { ...state };
+  const arrOfClones = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, { ...action.extraData });
+        break;
+
+      case 'removeProperties':
+        const values = action.keysToRemove;
+
+        values.map(el => delete stateCopy[el]);
```

Comment:
`map` is used when you need to gain a new modified array, in this case you only iterate throw an array, so better to use `forEach` or `forof`